### DISCUSSION TITLE
How to create the Set Group

### DISCUSSION THREAD

**Discussion by Ali141272** [neperfepx/neper#467] [Q&A] [answered]
I built the model, which you can find in the attached file. Here my main query is how to create the set group (Physical and Discrete surface ) for the top and bottom line in the 2D model. I have these surfaces for the 3D model but I am unable to create them in 2D.
Looking for receiving kind response and suggestion.
<img width="1088" alt="set" src="https://user-images.githubusercontent.com/102291665/167260319-3ac0e718-acc9-4766-b942-289239a1a1d2.PNG">
<img width="1075" alt="set1" src="https://user-images.githubusercontent.com/102291665/167260322-6ec0f836-8d76-4b2d-b189-7decb43521f3.PNG">

**Comment by rquey** [comment, marked as answer]
They are written under `$NSets` in the msh files, see option [-nset](https://neper.info/doc/neper_m.html#cmdoption-nset) and the specifications of the [Mesh File (.msh)](https://neper.info/doc/fileformat.html#mesh-file-msh).

For 2D, you will need to update to version 4.4.2-16.

**Reply by Ali141272** [reply]
Thank you very much Respected Professor for your valuable time and positive response.
Professor I have updated version of Gmsh 4.9.5 is it okay for my 2D model?

**Reply by rquey** [reply]
I was talking about Neper, not Gmsh. https://neper.info/downloads.html#latest-version

**Reply by Ali141272** [reply]
Professor my Neper version is 4.4.1 is it okay?
![Neper](https://user-images.githubusercontent.com/102291665/167415605-f50638ec-890f-43fa-a6f0-bfe1ae6734fb.PNG)
